packages/types/src/tx/schema/shielded.ts

```diff
@@ -0,0 +1,86 @@
+import BN from "bn.js";
+import { ShieldedDataProps, ShieldedProps } from "../types";
+
+// A Zcash Transaction
+export class ShieldedTransferMsgValue {
+  txid: Uint8Array;
+  data: Uint8Array;
+
+  constructor(properties: ShieldedProps) {
+    this.txid = properties.txId;
+    this.data = properties.data;
+  }
+}
+
+export const ShieldedTransferMsgSchema = new Map([
```

Comment:
nit: let's evaluate again in near future if these would be just best living on Rust side, as the types likely anyway come from there and we get all these conversions automatically correct and for free.

Reply:
Unfortunately we can't pass in `u64` types from TS natively, hence the schema here. But yeah, we can revisit it.